File: FTP Client for WinForms using Ultimate FTP library/[C#]-FTP Client for WinForms using Ultimate FTP library/C#/Ftp/FtpSettingInfo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ClientSample.Ftp
{
    class FtpSettingInfo
    {
        public const string SmartPath = "SmartPath";
        public const string SendAbortSignals = "SendAbortSignals";
        public const string SendAborCommand = "SendAborCommand";
        public const string ChangeDirBeforeListing = "ChangeDirBeforeListing";
        public const string ChangeDirBeforeTransfer = "ChangeDirBeforeTransfer";
        public const string Compress = "Compress";

        public static void SaveConfig(SettingInfoBase settings)
        {
            #region Ftp
            settings.SaveProperty(SendAbortSignals);
            settings.SaveProperty(SendAborCommand);
            settings.SaveProperty(ChangeDirBeforeListing);
            settings.SaveProperty(ChangeDirBeforeTransfer);
            settings.SaveProperty(SmartPath);
            settings.SaveProperty(Compress);
            #endregion
        }

        public static void LoadConfig(SettingInfoBase settings)
        {
            #region FTP
            settings.LoadProperty(SendAbortSignals, true);
            settings.LoadProperty(SendAborCommand, 0);
            settings.LoadProperty(ChangeDirBeforeListing, true);
            settings.LoadProperty(ChangeDirBeforeTransfer, false);
            settings.LoadProperty(SmartPath, false);
            settings.LoadProperty(Compress, false);
            #endregion
        }
    }
}

File: FTP Client for WinForms using Ultimate FTP library/[VB]-FTP Client for WinForms using Ultimate FTP library/VB/Ftp/FtpSettingInfo.vb
Imports System.Text

Namespace ClientSample.Ftp
	Friend Class FtpSettingInfo
		Public Const SmartPath As String = "SmartPath"
		Public Const SendAbortSignals As String = "SendAbortSignals"
		Public Const SendAborCommand As String = "SendAborCommand"
		Public Const ChangeDirBeforeListing As String = "ChangeDirBeforeListing"
		Public Const ChangeDirBeforeTransfer As String = "ChangeDirBeforeTransfer"
		Public Const Compress As String = "Compress"

		Public Shared Sub SaveConfig(ByVal settings As SettingInfoBase)
'			#Region "Ftp"
			settings.SaveProperty(SendAbortSignals)
			settings.SaveProperty(SendAborCommand)
			settings.SaveProperty(ChangeDirBeforeListing)
			settings.SaveProperty(ChangeDirBeforeTransfer)
			settings.SaveProperty(SmartPath)
			settings.SaveProperty(Compress)
'			#End Region
		End Sub

		Public Shared Sub LoadConfig(ByVal settings As SettingInfoBase)
'			#Region "FTP"
			settings.LoadProperty(SendAbortSignals, True)
			settings.LoadProperty(SendAborCommand, 0)
			settings.LoadProperty(ChangeDirBeforeListing, True)
			settings.LoadProperty(ChangeDirBeforeTransfer, False)
			settings.LoadProperty(SmartPath, False)
			settings.LoadProperty(Compress, False)
'			#End Region
		End Sub
	End Class
End Namespace
